### DISCUSSION TITLE
Custom non-persisted field in generated classes

### DISCUSSION THREAD

**Discussion by PaulVipond** [simolus3/drift#1020] [General]
Hi, I've done a fair amount of googling, but still not sure if it's possible. Is it possible to add a non-persisted field to the generated classes which return table content? Something like this would be great.

```
class ContentMessages extends Table {
  TextColumn get id => text().nullable().clientDefault(() => Uuid().v4())();
  TextColumn get contentGroupId => text()();

  bool isFromAppUser;
}
```

I realise I could transform this into another object, but it's a fairly simple case for one extra column.

**Comment by simolus3** [comment]
No, this is not possible by design. Generated row classes should only hold one row from the table.

What you could do is define an extension on the generated class and add the property with an [expando](https://api.dart.dev/stable/2.10.5/dart-core/Expando-class.html):

```dart
Expando<bool> _isFromAppUserExp = Expando();

extension on ContentMessage {
  bool get isFromAppUser => _isFromAppUserExp[this];
  set isFromAppUser(bool value) => _isFromAppUserExp[this] = value;
}
```

**Reply by PaulVipond** [reply]
Thanks for the reply Simon. I'll give this a shot.
